src/GNDStk/Node/src/write.hpp

```diff
@@ -14,21 +14,28 @@ std::ostream &write(std::ostream &os, const int level = 0) const
    const std::string inext(indent*(level+1),' '); // next ...
 
    // write name
-   os << icurr << name << ":" << std::endl;
+   if (!(os << icurr << name << ":" << std::endl))
+      log::error("ostream << node.name returned with !ostream");
 
    // write metadata
-   for (const auto &meta : metadata)
-      if (!(os << inext << meta.first << ": " << meta.second << std::endl))
-         break;
+   if (os)
+      for (const auto &meta : metadata)
+         if (!(os << inext << meta.first << ": " << meta.second << std::endl)) {
+            log::error("ostream << node.metadata returned with !ostream");
+            break;
+         }
 
    // write children
-   for (const auto &c : children)
-      if (c && !c->write(os,level+1))
-         break;
+   if (os)
+      for (const auto &c : children)
+         if (c && !c->write(os,level+1)) {
+            log::error("ostream << node.children returned with !ostream");
+            break;
+         }
 
-   // check for errors
+   // context, if applicable
    if (!os)
-      log::error("Problem during Node::write(ostream)");
+      log::member("Node.write(ostream)");
```

Comment:
The logging library actually provides namespace Log (capital L), and provides debug, info, warning, and error. (I think it can be extended to have other things.) I actually put some wrappers to the library into GNDStk's log (lower-case ell), and use those. Apparently Log::error() isn't throwing, because I seem to get to my log::* statements just beneath where I call my log::error() (which ultimately calls Log::error().) My log::member() basically identifies what it prints as being a member function, as opposed to, say, log::function() (a regular function, not a member function) and others. The intent here is to provide good, informative diagnostics, and in particular context. As in, "here's the error, and then here's the member function (or regular function (or assignment operator (or constructor (or ...)))) where it happened. It's easy to do, the code is easy to understand, and users will thank us for giving them good diagnostics.